File: backend/src/main/java/com/pgvaale/backend/entity/Menu.java
package com.pgvaale.backend.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import lombok.*;

import java.time.LocalDate;

/** MENU ENTITY */
@Entity
@Table(name = "menus")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Menu extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tiffin_id", nullable = false)
    private Tiffin tiffin;

    @NotBlank
    private String dayOfWeek; // Monday, Tuesday, etc.

    @NotBlank
    private String breakfast;

    @NotBlank
    private String lunch;

    @NotBlank
    private String dinner;

    private LocalDate menuDate; // Specific date for this menu

    private String foodCategory; // Veg, Non-Veg, Both

    private Double price; // Price per meal

    @Builder.Default
    private boolean isActive = true;
} 

File: PGVaaleDotNetBackend/Entities/Menu.cs
using System.ComponentModel.DataAnnotations;

namespace PGVaaleDotNetBackend.Entities
{
    public class Menu : BaseEntity
    {
        [Key]
        public long Id { get; set; }

        [Required]
        public long TiffinId { get; set; }

        [Required]
        public string DayOfWeek { get; set; } = string.Empty; // Monday, Tuesday, etc.

        [Required]
        public string Breakfast { get; set; } = string.Empty;

        [Required]
        public string Lunch { get; set; } = string.Empty;

        [Required]
        public string Dinner { get; set; } = string.Empty;

        public DateTime MenuDate { get; set; } // Specific date for this menu

        public string FoodCategory { get; set; } = string.Empty; // Veg, Non-Veg, Both

        public double Price { get; set; } // Price per meal

        public bool IsActive { get; set; } = true;

        // Navigation property
        public Tiffin Tiffin { get; set; } = null!;
    }
}
